### PR TITLE
clarify beans.xml handling in CDI Lite

### PR DESCRIPTION
**Pull request by Ladicek**

### CODE REVIEW

**Comment by Ladicek** [comment]
This is my proposal for #336.

File: spec/src/main/asciidoc/core/packagingdeployment.asciidoc [hunk 1]
```diff
@@ -38,6 +38,9 @@ An _implicit bean archive_ is:
 Any other archive which contains a `beans.xml` file is not portable in {cdi_lite}.
 More kinds of bean archives exist in {cdi_full}.
 
+Implementations that do not support {cdi_full} are required to ignore the content of the `beans.xml` file, except for the `bean-discovery-mode` attribute.
+Implementations that do not support {cdi_full} are required to detect presence of an archive which contains a `beans.xml` file that has `bean-discovery-mode` attribute set to `all` and treat it as a deployment problem.
```
[the remark this answers is not in the document]
Comment (manovotn) [reply spec/src/main/asciidoc/core/packagingdeployment.asciidoc:42]: Or perhaps we could water it down (just to make it clearer it's not being asserted anywhere) and say:
> Implementations that do not support {cdi_full} are encouraged/expected to...

**Review by manovotn — COMMENTED** [review]
I am fine specifying it like this so long as all interested parties acknowledge that while we spec it, we still cannot test it with TCKs.

File: spec/src/main/asciidoc/core/packagingdeployment.asciidoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (Emily-Jiang) [reply spec/src/main/asciidoc/core/packagingdeployment.asciidoc:42]: How about rewording something like:
Implementations that do not support {cdi_full} are required to detect presence of an archive which contains a `beans.xml` file that has `bean-discovery-mode` attribute. The attribute value neither `none` nor `annotated` can be treated as a deployment problem.

[the remark this answers is not in the document]
Comment (manovotn) [reply spec/src/main/asciidoc/core/packagingdeployment.asciidoc:42]: During the CDI meeting we agreed that we want to state this as `required` even though it cannot currently be tested with TCKs.
With that in mind, I think we can go ahead and merge this PR in its current form.

**Review by manovotn — APPROVED** [review]

**Review by graemerocher — APPROVED** [review]
